buildchain/buildchain/docker_command.py

```diff
@@ -0,0 +1,300 @@
+# coding: utf-8
+
+"""
+Expose Docker commands (build, run, …) as Python classes, using Docker API.
+
+The instantiated objects are callable and can be used as doit action directly.
+"""
+
+import copy
+import os
+from pathlib import Path
+from typing import Any, Dict, List, Optional
+
+import docker                            # type: ignore
+from docker.types import Mount           # type: ignore
+import doit                              # type: ignore
+from doit.exceptions import TaskError    # type: ignore
+
+from buildchain import constants
+
+
+DOCKER_CLIENT : docker.DockerClient = docker.from_env()
+
+
+# The call method is not counted as a public method
+# pylint: disable=too-few-public-methods
+class DockerBuild:
+    """A class to expose the `docker build` command through the API client."""
+
+    def __init__(
+        self,
+        tag: str,
+        path: Path,
+        dockerfile: Path,
+        buildargs: Dict[str, Any]
+    ):
+        """Initialize a `docker tag` callable object.
```

Comment:
Not entirely sure what you meant, but it seems more obvious to me to say "this object is used to build a container image, and has some metadata (e.g. a tag)" rather than "this object is used to tag a container, with the metadata necessary to build the container first".

```suggestion
        """Initialize a `docker build` callable object.
```

Reply:
Oh actually the `docker tag` is below